cocos/core/platform/visible-rect.ts

```diff
@@ -33,8 +33,9 @@ import { Rect } from '../math';
 import { legacyCC } from '../global-exports';
 
 /**
- * `visibleRect` is a singleton object which defines the actual visible rect of the current view,
- * it should represent the same rect as `view.getViewportRect()`
+ * `visibleRect` is a singleton object which defines the actual visible rect of the current view
+ *
+ * @inner please use view.getVisibleSize() instead.
```

Comment:
I have take a look at getVisibleSize, it's using `new Size` internally, please update it to return `visibleRect` directly.